tests/dash/gnmi_utils.py

```diff
@@ -0,0 +1,377 @@
+    time.sleep(env.gnmi_server_start_wait_time)
+
+
+def gnmi_set(duthost, ptfhost, delete_list, update_list, replace_list):
+    env = GNMIEnvironment(duthost)
+    ip = duthost.mgmt_ip
+    port = env.gnmi_port
+    cmd = 'python2 /root/gnxi/gnmi_cli_py/py_gnmicli.py '
+    cmd += '--timeout 30 '
+    cmd += '-t %s -p %u ' % (ip, port)
+    cmd += '-xo sonic-db '
+    cmd += '-rcert /root/%s ' % (env.gnmi_ca_cert)
+    cmd += '-pkey /root/%s ' % (env.gnmi_client_key)
+    cmd += '-cchain /root/%s ' % (env.gnmi_client_cert)
+    cmd += '-m set-update '
+    xpath = ''
+    xvalue = ''
+    for path in delete_list:
+        path = path.replace('sonic-db:', '')
+        xpath += ' ' + path
+        xvalue += ' ""'
+    for update in update_list:
+        update = update.replace('sonic-db:', '')
+        result = update.rsplit(':', 1)
+        xpath += ' ' + result[0]
+        xvalue += ' ' + result[1]
+    for replace in replace_list:
+        replace = replace.replace('sonic-db:', '')
+        result = replace.rsplit(':', 1)
+        xpath += ' ' + result[0]
+        if '#' in result[1]:
+            xvalue += ' ""'
+        else:
+            xvalue += ' ' + result[1]
+    cmd += '--xpath ' + xpath
+    cmd += ' '
+    cmd += '--value ' + xvalue
+    output = ptfhost.shell(cmd, module_ignore_errors=True)
+    error = "GRPC error\n"
+    if error in output['stdout']:
+        result = output['stdout'].split(error, 1)
+        return -1, result[1]
+    if output['stderr']:
+        return -1, output['stderr']
+    else:
+        return 0, output['stdout']
+
+
+def gnmi_get(duthost, ptfhost, path_list):
+    env = GNMIEnvironment(duthost)
+    ip = duthost.mgmt_ip
+    port = env.gnmi_port
+    cmd = 'python2 /root/gnxi/gnmi_cli_py/py_gnmicli.py '
+    cmd += '--timeout 30 '
+    cmd += '-t %s -p %u ' % (ip, port)
+    cmd += '-xo sonic-db '
+    cmd += '-rcert /root/%s ' % (env.gnmi_ca_cert)
+    cmd += '-pkey /root/%s ' % (env.gnmi_client_key)
+    cmd += '-cchain /root/%s ' % (env.gnmi_client_cert)
+    cmd += '--encoding 4 '
+    cmd += '-m get '
+    cmd += '--xpath '
+    for path in path_list:
+        path = path.replace('sonic-db:', '')
+        cmd += " " + path
+    output = ptfhost.shell(cmd, module_ignore_errors=True)
+    if output['stderr']:
+        return -1, [output['stderr']]
+    else:
+        msg = output['stdout'].replace('\\', '')
+        error = "GRPC error\n"
+        if error in msg:
+            result = msg.split(error, 1)
+            return -1, [result[1]]
+        mark = 'The GetResponse is below\n' + '-'*25 + '\n'
+        if mark in msg:
+            result = msg.split(mark, 1)
+            msg_list = result[1].split('-'*25)[0:-1]
+            return 0, [msg.strip("\n") for msg in msg_list]
+        else:
+            return -1, [msg]
+
+
+def json_to_proto(key, json_obj):
+    table_name = re.search(r"DASH_(\w+)_TABLE", key).group(1)
+    if table_name == "APPLIANCE":
+        pb = Appliance()
```

Comment:
I suggest splitting the implementation of json_to_proto and making it more generic, so it will be easier to add support for the new objects in the future:
```
def appliance_from_json(json_obj):
    pb = Appliance()
    pb.sip.ipv4 = socket.htonl(int(ipaddress.IPv4Address(json_obj["sip"])))
    pb.vm_vni = int(json_obj["vm_vni"])
    return pb

def vnet_from_json(json_obj):
    pb = Vnet()
    pb.vni = int(json_obj["vni"])
    pb.guid.value = bytes.fromhex(uuid.UUID(json_obj["guid"]).hex)
    return pb

handlers_map = {
    "DASH_APPLIANCE": appliance_from_json,
    "DASH_VNET": vnet_from_json
}

def json_to_proto(key, json_obj):
    table, row = key.split(TABLE_NAME_SEPARATOR, 1)
    if table not in handlers_map:
        raise RuntimeError(f"Unknown DASH table: {table}")

    return handlers_map[table](json_obj)
```

Reply:
Done